### PR TITLE
Feat/add component tests

### PR DESCRIPTION
**Pull request by Thistle-Strum**
🔥 Summary
-Draft Tests for Button, CustomSelect, SearchSelect and Navbar Components.  Also, adds chevron animation for `select` 

😤 Problem / Goals
-SearchSelect and Navbar tests are not yet complete.  I need to dig deeper into how the react-responsive package interacts with these two components.  Any insight is much appreciated! 

 🤓 Solution
-Continue to focus on implementing react-responsive in SearchSelect and Navbar tests

### CODE REVIEW

File: frontend/src/components/base/Button/Button.test.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import { fireEvent, render } from '../../../test-utils';
+import { Button } from './Button';
+
+describe('Button', () => {
+  it('should render without errors', () => {
```
Comment (DorianKwan) [inline frontend/src/components/base/Button/Button.test.tsx]: This should also be fine; no need to change just FYI

```suggestion
  it('should render', () => {
```

File: frontend/src/components/base/Button/Button.test.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    const { getByRole } = render(<Button type="button">button</Button>);
+    const button = getByRole('button', { name: 'button' });
+
+    expect(button).toBeInTheDocument();
+  });
+
+  it('should produce an event on click if not disabled', () => {
```
Comment (DorianKwan) [inline frontend/src/components/base/Button/Button.test.tsx]: I like that you're trying to test that the button click event works.  But I'm not sure I understand how exactly it's testing that.  Isn't the `.toBeEnabled` method checking whether or not the button is not disabled

File: frontend/src/components/base/Button/Button.tsx [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+      {children}
+    </StyledButton>
+  );
+};
+
+const StyledButton = styled.button<{ bgColor: string }>`
+  color: white;
+  background-color: ${({ bgColor }) => bgColor};
+  text-align: center;
+  padding: 10px;
```
Comment (DorianKwan) [inline frontend/src/components/base/Button/Button.tsx]: Can we use rem here?

File: frontend/src/components/buttons/NavButton.tsx [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        <Button type={type} onClick={onClick} color={color}>
+          {children}
+        </Button>
+      </ButtonStyles>
+    </NavButtonContainer>
+  );
+};
+
+export const NavButtonContainer = styled.div`
+  z-index: 30;
```
Comment (DorianKwan) [inline frontend/src/components/buttons/NavButton.tsx]: Can you expand on why we need a z-index here?

File: frontend/src/components/layout/Header/Partials/CustomSelect/CustomSelect.tsx [hunk 5]
```diff
@@ -0,0 +1,152 @@
+import styled from '@emotion/styled';
+import React from 'react';
+import Select, { PropsValue } from 'react-select';
+import { pxToRem, colors } from '../../../../../styled-theme';
+import { SelectOptions } from '../partials.types';
+
+export interface CustomSelectProps {
+  readonly isMenuOpen: boolean;
+  readonly setIsMenuOpen: (value: React.SetStateAction<boolean>) => void;
```
Comment (DorianKwan) [inline frontend/src/components/layout/Header/Partials/CustomSelect/CustomSelect.tsx]: Could this just be state within this component?

File: frontend/src/components/layout/Header/Partials/SearchSelect/SearchSelect.test.tsx [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+import { FormValues } from '../partials.types';
+
+// window.resizeTo = function resizeTo(width, height): void {
+//   Object.assign(this, {
+//     innerWidth: width,
+//     innerHeight: height,
+//     outerWidth: width,
+//     outerHeight: height,
+//   }).dispatchEvent(new this.Event('resize'));
+// };
```
Comment (DorianKwan) [inline frontend/src/components/layout/Header/Partials/SearchSelect/SearchSelect.test.tsx]:
```suggestion
```

File: frontend/src/components/layout/Navbar/Navbar.test.tsx [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+  it('should hide navigation options when screen width is below 1023px', () => {
+    const { container: mobile } = render(
+      // eslint-disable-next-line react/jsx-no-constructed-context-values
+      <ResponsiveContext.Provider value={{ width: 1022 }}>
+        <Navbar />
+      </ResponsiveContext.Provider>,
+    );
+
+    // const navItem = getByText('Home');
+    // expect(navItem).not.toBeInTheDocument();
```
Comment (DorianKwan) [inline frontend/src/components/layout/Navbar/Navbar.test.tsx]:
```suggestion
```

File: frontend/src/components/buttons/NavButton.tsx (hunk 4 above)
Comment (DorianKwan) [reply frontend/src/components/buttons/NavButton.tsx]: I noticed this was moved around  also; perhaps we don't need it anymore?

File: frontend/src/components/layout/Navbar/Navbar.tsx [hunk 8]
```diff
@@ -55,13 +55,14 @@ export const Navbar: React.FC = () => {
   }, [isOpened]);
 
   return (
-    <Nav>
+    <Nav aria-label="navigation">
```
Comment (DorianKwan) [inline frontend/src/components/layout/Navbar/Navbar.tsx]: The semantic tag here is enough for accessibility here.  Is there another reason you're adding this?

```suggestion
    <Nav>
```

**Review by DorianKwan — CHANGES_REQUESTED** [review]
Nice work with the tests.  A few comments here or there. But great to see you dive deeper into testing!

File: frontend/src/components/base/Button/Button.test.tsx (hunk 1 above)
Comment (Thistle-Strum) [reply frontend/src/components/base/Button/Button.test.tsx]: Short and concise! Done.

File: frontend/src/components/base/Button/Button.test.tsx (hunk 2 above)
Comment (Thistle-Strum) [reply frontend/src/components/base/Button/Button.test.tsx]: Yeah, this is something I intended to address. Adding a mock callback function onClick seems to have done the trick.

File: frontend/src/components/base/Button/Button.tsx (hunk 3 above)
Comment (Thistle-Strum) [reply frontend/src/components/base/Button/Button.tsx]: Done.

File: frontend/src/components/buttons/NavButton.tsx (hunk 4 above)
Comment (Thistle-Strum) [reply frontend/src/components/buttons/NavButton.tsx]: Z-index is needed for the the `MobileNavItemsContainer` (ie. `background-color: Z-index: 10` and CloseMenuButton `z-index: 20`).  `MobileNavItemsContainer` needs more refactoring.  Specifically, a way to disable x-scrolling.  I've added it to our Notion board.

File: frontend/src/components/layout/Header/Partials/CustomSelect/CustomSelect.tsx (hunk 5 above)
Comment (Thistle-Strum) [reply frontend/src/components/layout/Header/Partials/CustomSelect/CustomSelect.tsx]: For sure!  Done.

File: frontend/src/components/layout/Header/Partials/SearchSelect/SearchSelect.test.tsx (hunk 6 above)
Comment (Thistle-Strum) [reply frontend/src/components/layout/Header/Partials/SearchSelect/SearchSelect.test.tsx]: Done.

File: frontend/src/components/layout/Navbar/Navbar.test.tsx (hunk 7 above)
Comment (Thistle-Strum) [reply frontend/src/components/layout/Navbar/Navbar.test.tsx]: Done.

File: frontend/src/components/layout/Navbar/Navbar.tsx (hunk 8 above)
Comment (Thistle-Strum) [reply frontend/src/components/layout/Navbar/Navbar.tsx]: Currently, I am using this label for a `roleMatcher` in the test file.

File: frontend/src/components/buttons/NavButton.tsx (hunk 4 above)
Comment (DorianKwan) [reply frontend/src/components/buttons/NavButton.tsx]: Ahhh okay, makes sense.  Good work man, thanks for the clarification

File: frontend/src/components/layout/Navbar/Navbar.tsx (hunk 8 above)
Comment (DorianKwan) [reply frontend/src/components/layout/Navbar/Navbar.tsx]: Ahh okay, can we update this to use `data-testid`?  I've done this in the past where I would use aria tags for testing lookups, but I don't believe that's the best solution

File: frontend/src/components/layout/Header/Partials/CustomSelect/CustomSelect.tsx (hunk 5 above)
Comment (DorianKwan) [reply frontend/src/components/layout/Header/Partials/CustomSelect/CustomSelect.tsx]: I don't think you've pushed your changes for this yet

File: frontend/src/components/layout/Header/Partials/SearchSelect/SearchSelect.test.tsx (hunk 6 above)
Comment (DorianKwan) [reply frontend/src/components/layout/Header/Partials/SearchSelect/SearchSelect.test.tsx]: Same here

File: frontend/src/components/layout/Navbar/Navbar.test.tsx (hunk 7 above)
Comment (DorianKwan) [reply frontend/src/components/layout/Navbar/Navbar.test.tsx]: And here

File: frontend/src/components/base/Button/Button.test.tsx (hunk 2 above)
Comment (DorianKwan) [reply frontend/src/components/base/Button/Button.test.tsx]: Ahh yes that would do the trick.  That said, I believe there is a better approach.  Please see my updated comment :)

File: frontend/src/components/base/Button/Button.test.tsx [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+      <Button type="button" onClick={mockClickHandler} isDisabled>
+        button
+      </Button>,
+    );
+    const button = getByRole('button', { name: 'button' });
+
+    userEvent.click(button);
+
+    expect(button).not.toBeEnabled();
+    expect(mockClickHandler.mock.calls.length).toEqual(0);
```
Comment (DorianKwan) [inline frontend/src/components/base/Button/Button.test.tsx]: Sweet, a check like this is exactly what we needed.

But, are you up to update it to this?

```suggestion
    expect(mockClickHandler).toHaveBeenCalledTimes(1);
```

But add the jest function within this it

File: frontend/src/components/layout/Header/Partials/CustomSelect/CustomSelect.tsx (hunk 5 above)
Comment (DorianKwan) [reply frontend/src/components/layout/Header/Partials/CustomSelect/CustomSelect.tsx]: Awesome it's there now

File: frontend/src/components/base/Button/Button.test.tsx (hunk 9 above)
Comment (Thistle-Strum) [reply frontend/src/components/base/Button/Button.test.tsx]: Done.

File: frontend/src/components/layout/Navbar/Navbar.tsx (hunk 8 above)
Comment (Thistle-Strum) [reply frontend/src/components/layout/Navbar/Navbar.tsx]: Done.

File: frontend/src/components/base/Button/Button.test.tsx [hunk 10]
```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import userEvent from '@testing-library/user-event';
+import { render } from '../../../test-utils';
+import { Button } from './Button';
+
+describe('Button', () => {
+  const mockClickHandler = jest.fn();
```
Comment (DorianKwan) [inline frontend/src/components/base/Button/Button.test.tsx]: Can you move this into the `it`'s where it's being used?

**Review by DorianKwan — CHANGES_REQUESTED** [review]

File: frontend/src/components/base/Button/Button.test.tsx (hunk 10 above)
Comment (Thistle-Strum) [reply frontend/src/components/base/Button/Button.test.tsx]: Done.

**Review by DorianKwan — APPROVED** [review]

**Pull request merged**